bf2fc6cc711aee1a0c2a/kas-fleet-manager
# add config for enterprise credentials and reconcile logging credentials per org

## Description

For enterprise clusters we only want to allow access to a subset of logGroups in cloudwatch. This PR adds the ability to specify, for enterprise clusters, a separate IAM credential for logging that will only reconcile to a cluster belonging to the org named in that credential config.

The enterprise
## Verification Steps

_verify enterprise_
1. Setup a development OSD cluster and register it with KFM as an enterprise cluster with an orgId. I was using these commands in the db
```
update clusters set cluster_type='enterprise';
update clusters set organization_id='<THE_ORG_ID>';
```

2. Create the observability-cloudwatch-logs.yaml file under secrets/ with a content like this
```
---
aws_enterprise_iam_credentials:
  - org_id: <Actual_org_id_of_cluster>
    aws_access_key: anaenterpriseccesskeyfromorg
    aws_secret_access_key: anenterprisesecretaccesskeyfromorg
  - org_id: anorgid2
    aws_access_key: anotherenterprisekey
    aws_secret_access_key: anotherenterprisekey

aws_iam_credentials:
  aws_access_key: anaccesskey
  aws_secret_access_key: asecretaccesskey
```
3. Run KFM and verify the secret has been added to the syncset with the correct values.

_verify non-enterprise_
1. Teardown db and do not register cluster as enterprise
2. Verify that the default credentials are correctly added when the cluster is not enterprise.

_verify empty_
1. Change the cloudwatch config to an empty array:
```
---
aws_enterprise_iam_credentials []

aws_iam_credentials:
  aws_access_key: anaccesskey
  aws_secret_access_key: asecretaccesskey
```
2. Verify that enterprise clusters have an empty secret added the the syncset
3. Verify that non-enterprise clusters have the correct credentials in the syncset


## Checklist (Definition of Done)
